pkg/util/client.go

```diff
@@ -48,35 +49,43 @@ func LoadKubeConfig(path string) (*rest.Config, error) {
 
 // LoadKubeConfigs loads kubeconfigs. If the kubeconfigChangedCallBack is non-nil, it will watch all kubeconfigs it loaded
 // and call the callback once they change.
-func LoadKubeConfigs(kubeconfig string, kubeconfigChangedCallBack func(fsnotify.Event)) (map[string]*rest.Config, string, error) {
-	loader := clientcmd.NewDefaultClientConfigLoadingRules()
-	loader.ExplicitPath = kubeconfig
-	cfg, err := loader.Load()
+func LoadKubeConfigs(kubeconfig, kubeconfigDir string, kubeconfigChangedCallBack func(fsnotify.Event)) (map[string]*rest.Config, error) {
+	var errs []error
+	configs, err := kube.LoadClusterConfigs(kube.NewConfig(kube.ConfigFile(kubeconfig),
```

Comment:
Do we still load the configs from env if `KUBECONFIG` is set?

Reply:
If both `kubeconfig` and `kubeconfigDir` are empty, it will loads `$KUBECONFIG`.

https://github.com/kubernetes/test-infra/blob/93ddf012c30d14168b62a9e9dcffd8f546b1d7f6/prow/kube/config.go#L117

But if we found file, for example, from `kubeconfigDir`, then `$KUBECONFIG` is ignored.
Comparing to previous implementation, this does not change.

Reply:
In our use case, we either set $KUBECONFIG or use kubeconfigDir, not both, right?